client/lcd/test_helpers.go

```diff
@@ -408,3 +415,851 @@ func Request(t *testing.T, port, method, path string, payload []byte) (*http.Res
+}
+
+// PUT /keys/{name} Update the password for this account in the KMS
+func updateKey(t *testing.T, port, name, oldPassword, newPassword string, fail bool) {
+	kr := updateKeyReq{oldPassword, newPassword}
+	req, err := cdc.MarshalJSON(kr)
+	require.NoError(t, err)
+	keyEndpoint := fmt.Sprintf("/keys/%s", name)
+	res, body := Request(t, port, "PUT", keyEndpoint, req)
+	if fail {
+		require.Equal(t, http.StatusUnauthorized, res.StatusCode, body)
+		return
+	}
+	require.Equal(t, http.StatusOK, res.StatusCode, body)
+}
+
+type updateKeyReq struct {
+	OldPassword string `json:"old_password"`
+	NewPassword string `json:"new_password"`
+}
+
+// DELETE /keys/{name} Remove an account
+func deleteKey(t *testing.T, port, name, password string) {
+	dk := deleteKeyReq{password}
+	req, err := cdc.MarshalJSON(dk)
+	require.NoError(t, err)
+	keyEndpoint := fmt.Sprintf("/keys/%s", name)
+	res, body := Request(t, port, "DELETE", keyEndpoint, req)
+	require.Equal(t, http.StatusOK, res.StatusCode, body)
+}
+
+type deleteKeyReq struct {
+	Password string `json:"password"`
+}
+
+// GET /auth/accounts/{address} Get the account information on blockchain
+func getAccount(t *testing.T, port string, addr sdk.AccAddress) auth.Account {
+	res, body := Request(t, port, "GET", fmt.Sprintf("/auth/accounts/%s", addr.String()), nil)
+	require.Equal(t, http.StatusOK, res.StatusCode, body)
+	var acc auth.Account
+	err := cdc.UnmarshalJSON([]byte(body), &acc)
+	require.Nil(t, err)
+	return acc
+}
+
+// ----------------------------------------------------------------------
+// ICS 20 - Tokens
+// ----------------------------------------------------------------------
+
+// TODO: TEST THE FOLLOWING ROUTES
+// POST /tx/sign Sign a Tx
+// POST /tx/broadcast Send a signed Tx
+// GET /bank/balances/{address} Get the account balances
+
+// POST /bank/accounts/{address}/transfers Send coins (build -> sign -> send)
+func doSend(t *testing.T, port, seed, name, password string, addr sdk.AccAddress) (receiveAddr sdk.AccAddress, resultTx ctypes.ResultBroadcastTxCommit) {
+	res, body, receiveAddr := doSendWithGas(t, port, seed, name, password, addr, "", 0, false, false)
+	require.Equal(t, http.StatusOK, res.StatusCode, body)
+
+	err := cdc.UnmarshalJSON([]byte(body), &resultTx)
+	require.Nil(t, err)
+
+	return receiveAddr, resultTx
+}
+
+func doSendWithGas(t *testing.T, port, seed, name, password string, addr sdk.AccAddress, gas string,
+	gasAdjustment float64, simulate, generateOnly bool) (
+	res *http.Response, body string, receiveAddr sdk.AccAddress) {
+
+	// create receive address
+	kb := client.MockKeyBase()
+	receiveInfo, _, err := kb.CreateMnemonic("receive_address", cryptoKeys.English, gapp.DefaultKeyPass, cryptoKeys.SigningAlgo("secp256k1"))
+	require.Nil(t, err)
+	receiveAddr = sdk.AccAddress(receiveInfo.GetPubKey().Address())
+
+	acc := getAccount(t, port, addr)
+	accnum := acc.GetAccountNumber()
+	sequence := acc.GetSequence()
+	chainID := viper.GetString(client.FlagChainID)
```

Comment:
uhh can we change this everywhere to use `sdk.DefaultChainID()` instead ?

Reply:
One potential refactor is to rewrite the `InitializeTestLCD` to return a `Fixtures` object with all this stuff stored there, but I would like to get this change merged first.